src/clients/asset-client.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const Client = require('./base-client')
+const simpleFetch = require('../utils/simple-fetch')
+const mime = require('mime-types')
+
+class AssetClient {
+  constructor (url) {
+    this.client = new Client({
+      name: 'AssetClient',
+      url: url,
+      contentType: 'application/vnd.api+json'
+    })
+  }
+
+  create (params) {
+    return this.__createOrUpdate(params)
+  }
+
+  destroy (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/assets/${id}`, {
+      method: 'DELETE'
+    })
+  }
+
+  index () {
+    return this.client.sendRequest(`${this.client.baseUrl}/assets`)
+  }
+
+  show (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/assets/${id}`)
+  }
+
+  update (params) {
+    return this.__createOrUpdate(params)
+  }
+
+  download (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/${id}`)
+  }
+
+  display (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/${id}/inline`)
+  }
+
+  __createOrUpdate (params) {
+    const id = params.id
+    const data = params.data
+    const file = params.file
+
+    const url = id ? `${this.client.baseUrl}/assets/${id}` : `${this.client.baseUrl}/assets`
+    const method = id ? 'PUT' : 'POST'
```

Comment:
I think we can get rid of some logic in here, if we just remove these bits and pass them in as params, i. e.

```jsx

create (params) {
  this.__createOrUpdate('POST', `${this.client.baseUrl}/assets`, params)
}

update (params) {
  this.__createOrUpdate('PUT', `${this.client.baseUrl}/assets/${params.id}`, params)
}

__createOrUpdate (method, url, params) {
  // Snip.
}
```

Reply:
I like that!